### PR TITLE
Fix UEFI detection on KVM and prevent deployments on non UEFI enabled hosts

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR fixes 2 issues:
- Prevent UEFI deployments on non UEFI enabled hosts (do not assume the host detail will always be present in case of not enabled)
- Fix the UEFI detection on KVM (uefi.properties file check was missing)

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?
Tested on KVM - one UEFI enabled host, one disabled

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Review by weizhouapache — APPROVED** [review]
code lgtm

**Comment by nvazquez** [comment]
@blueorangutan test

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java [hunk 1]
```diff
@@ -1414,6 +1414,10 @@ private void configureAgentHooks(final Map<String, Object> params) {
         s_logger.debug("agent.hooks.libvirt_vm_on_stop.method is " + _agentHooksVmOnStopMethod);
     }
 
+    public boolean isUefiPropertiesFileSet() {
```
Comment (sureshanaparti) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:1417]: @nvazquez UEFI properties are loaded from this file while configuring the resource, no need to check again for this file, instead check for properties loaded or not. Also, re-use this method in _loadUefiProperties()_ method below for the same condition. Thanks.

```suggestion
    public boolean isUefiPropertiesLoaded() {
        return _uefiProperties != null && _uefiProperties.getProperty("guest.loader.legacy") != null
    }
```

Comment (nvazquez) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java]: Done, thanks - nice catch

**Comment by nvazquez** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Review by slavkap — APPROVED** [review]
code looks good

**Pull request merged**